compiler/noirc_evaluator/src/ssa/interpreter/mod.rs

```diff
@@ -736,11 +736,13 @@ impl<'ssa> Interpreter<'ssa> {
         &mut self,
         array: ValueId,
         index: ValueId,
+        offset: ArrayGetOffset,
         result: ValueId,
     ) -> IResult<()> {
         let element = if self.side_effects_enabled() {
             let array = self.lookup_array_or_slice(array, "array get")?;
             let index = self.lookup_u32(index, "array get index")?;
+            let index = index - offset.to_u32();
```

Comment:
It just occurred to me we don't check for index out of bounds (or underflow now) here. I'll make an issue for it.